qiskit/algorithms/eigensolvers/ssvqe.py

```diff
@@ -0,0 +1,575 @@
+            operator: The operator whose energy to evaluate.
+        Returns:
+            A function handle to evaluate the gradient at given parameters for the initialized
+            ansatz list.
+        Raises:
+            AlgorithmError: If the primitive job to evaluate the gradient fails.
+        """
+
+        def evaluate_gradient(parameters):
+            # broadcasting not required for the estimator gradients
+            try:
+                job = self.gradient.run(
+                    initialized_ansatz_list,
+                    [operator] * self.k,
+                    [parameters for n in range(self.k)],
+                )
+                energy_gradients = job.result().gradients
+                weighted_energy_sum_gradient = sum(
+                    [self.weight_vector[n] * energy_gradients[n] for n in range(self.k)]
+                )
+            except Exception as exc:
+                raise AlgorithmError("The primitive job to evaluate the gradient failed!") from exc
+
+            return weighted_energy_sum_gradient
+
+        return evaluate_gradient
+
+    def _check_circuit_num_qubits(
+        self, operator: BaseOperator | PauliSumOp, circuit: QuantumCircuit, circuit_type: str
+    ) -> QuantumCircuit:
+        """Check that the number of qubits for the circuit passed matches
+        the number of qubits  of the operator.
+        """
+        if operator.num_qubits != circuit.num_qubits:
+            try:
+                logger.info(
+                    "Trying to resize %s to match operator on %s qubits.",
+                    circuit_type,
+                    operator.num_qubits,
+                )
+                circuit.num_qubits = operator.num_qubits
+            except AttributeError as error:
+                raise AlgorithmError(
+                    f"The number of qubits of the {circuit_type} does not match the ",
+                    f"operator, and the {circuit_type} does not allow setting the "
+                    "number of qubits using `num_qubits`.",
+                ) from error
+        return circuit
+
+    def _check_operator_ansatz(self, operator: BaseOperator | PauliSumOp) -> QuantumCircuit:
+        """Check that the number of qubits of operator and ansatz match and that the ansatz is
+        parameterized.
+        """
+        # set defaults
+        if self.ansatz is None:
+            ansatz = RealAmplitudes(num_qubits=operator.num_qubits, reps=6)
+        else:
+            ansatz = self.ansatz
+
+        ansatz = self._check_circuit_num_qubits(
+            operator=operator, circuit=ansatz, circuit_type="ansatz"
+        )
+
+        if ansatz.num_parameters == 0:
+            raise AlgorithmError("The ansatz must be parameterized, but has no free parameters.")
+
+        return ansatz
+
+    def _check_operator_initial_states(
+        self, list_of_states: Sequence[QuantumCircuit] | None, operator: BaseOperator | PauliSumOp
+    ) -> QuantumCircuit:
```

Comment:
Shall we actually have `list[QuantumCircuit]`?

Reply:
I agree, I have now changed this.